qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -0,0 +1,431 @@
+    each excited state to be orthogonal to all the previous excited states.
+
+    An optional array of parameter values, via the *initial_point*, may be provided
+    as the starting point for the search of the minimum eigenvalue. This feature is
+    particularly useful when there are reasons to believe that the solution point
+    is close to a particular point.
+
+    The length of the *initial_point* list value must match the number of the parameters
+    expected by the ansatz. If the *initial_point* is left at the default
+    of ``None``, then VQD will look to the ansatz for a preferred value, based on its
+    given initial state. If the ansatz returns ``None``,
+    then a random point will be generated within the parameter bounds set, as per above.
+    If the ansatz provides ``None`` as the lower bound, then VQD
+    will default it to :math:`-2\pi`; similarly, if the ansatz returns ``None``
+    as the upper bound, the default value will be :math:`2\pi`.
+
+    The following attributes can be set via the initializer but can also be read and
+    updated once the VQD object has been constructed.
+
+    Attributes:
+            estimator (BaseEstimator): The primitive instance used to perform the expectation
+                estimation as indicated in the VQD paper.
+            fidelity (BaseStateFidelity): The fidelity class instance used to compute the
+                overlap estimation as indicated in the VQD paper.
+            ansatz (QuantumCircuit): A parameterized circuit used as ansatz for the wave function.
+            optimizer(Optimizer): A classical optimizer. Can either be a Qiskit optimizer or a callable
+                that takes an array as input and returns a Qiskit or SciPy optimization result.
+            k (int): the number of eigenvalues to return. Returns the lowest k eigenvalues.
+            betas (list[float]): beta parameters in the VQD paper.
+                Should have length k - 1, with k the number of excited states.
+                These hyper-parameters balance the contribution of each overlap term to the cost
+                function and have a default value computed as the mean square sum of the
+                coefficients of the observable.
+            callback (Callable[[int, np.ndarray, float, dict[str, Any]], None] | None):
+                a callback that can access the intermediate data
+                during the optimization. Four parameter values are passed to the callback as
+                follows during each evaluation by the optimizer: the evaluation count,
+                the optimizer parameters for the ansatz, the evaluated mean, the evaluation
+                metadata, and the current step.
+    """
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        fidelity: BaseStateFidelity,
+        ansatz: QuantumCircuit,
+        optimizer: Optimizer | Minimizer,
+        *,
+        k: int = 2,
+        betas: list[float] | None = None,
```

Comment:
Can we make this a Sequence[float] to be less specific than the list type, like is done for initial_point